modules/osm/entity.js

```diff
@@ -3,40 +3,51 @@ import _keys from 'lodash-es/keys';
 import _toPairs from 'lodash-es/toPairs';
 import _union from 'lodash-es/union';
 import _without from 'lodash-es/without';
+import _find from 'lodash-es/find';
 
 import { debug } from '../index';
 import { osmIsInterestingTag } from './tags';
 import { dataDeprecated } from '../../data/index';
 
-
 export function osmEntity(attrs) {
     // For prototypal inheritance.
     if (this instanceof osmEntity) return;
 
     // Create the appropriate subtype.
     if (attrs && attrs.type) {
-        return osmEntity[attrs.type].apply(this, arguments);
-    } else if (attrs && attrs.id) {
-        return osmEntity[osmEntity.id.type(attrs.id)].apply(this, arguments);
-    }
 
+        var _activeLayer = _find(Hoot.layers.loadedLayers, function(a, b) { return a.activeLayer; });
+        if (_activeLayer && _activeLayer.id) {
```

Comment:
totally just a stylistic suggestion, so no need to change, but I almost wonder if we should make a new func for grabbing the ids from the active layer. something like `osmEntity.id.hoot` that looks a little like this...

```javascript
function(type, activeLayer) {
   return osmEntity.id.fromOsm(type, activeLayer.activeIds[type]--) + '_' + activeLayer.id;
}
```

Then here in this if block you just say "do I have an activeLayer'?
Ok I do, use `id.hoot` function
If I don't use the out of the box id way of doing things.

just a thought!